dhis-2/dhis-web-api/src/main/java/org/hisp/dhis/webapi/controller/SqlViewController.java

```diff
@@ -283,4 +274,109 @@ private GridResponse buildResponse( SqlView sqlView, SqlViewQuery query )
         }
         return new GridResponse( query.isSkipPaging() ? null : query.getPager(), grid );
     }
+
+    @Override
+    protected void preCreateEntity( SqlView entity )
+        throws ConflictException
+    {
+        checkJobConfigurationExists( entity );
+    }
+
+    @Override
+    protected void postCreateEntity( SqlView entity )
+    {
+        if ( entity.getUpdateJobId() != null )
+        {
+            addAndUpdateIfNeeded( entity );
+        }
+    }
+
+    @Override
+    protected void preUpdateEntity( SqlView entity, SqlView newEntity )
+        throws ConflictException
+    {
+        removeAndUpdateIfNeeded( newEntity );
+        if ( newEntity.getUpdateJobId() != null )
+        {
+            checkJobConfigurationExists( newEntity ); // validate
+            addAndUpdateIfNeeded( newEntity );
+        }
+    }
+
+    @Override
+    protected void prePatchEntity( SqlView entity, SqlView newEntity )
+        throws ConflictException
+    {
+        preUpdateEntity( entity, newEntity );
+    }
+
+    @Override
+    protected void preDeleteEntity( SqlView entity )
+    {
+        removeAndUpdateIfNeeded( entity );
+    }
+
+    private void checkJobConfigurationExists( SqlView entity )
+        throws ConflictException
+    {
+        String jobId = entity.getUpdateJobId();
+        if ( jobId == null )
+        {
+            return;
+        }
+        JobConfiguration config = jobConfigurationService.getJobConfigurationByUid( jobId );
```

Comment:
Could this null check be pushed to service layer through a `loadJobConfigurationJobByUid` method or similar?